src/proto/worker.proto

```diff
@@ -90,14 +91,15 @@ message ProtoHeader {
     optional bytes md5 = 3;
     optional string wname = 4;
     required bool largeresult = 5;
+    required bool endnodata = 6; // True if this header is the end, no more data. size should be 0.
+    optional uint32 seq = 7; // sequence number from SendChannel
+    optional int32 scsseq = 8; // sequence number from SendChannelShared, can be -1
 }
 
 message ColumnSchema {
     optional string name = 1; // Optional to allow type-only transmission
-    required bool deprecated_hasdefault = 2; // DEPRECATED (but 'required', so will have to continue to be set)
-    optional bytes deprecated_defaultvalue = 3; // DEPRECATED
-    required string sqltype = 4;
-    optional int32 mysqltype = 5;
+    required string sqltype = 2;
+    optional int32 mysqltype = 3;
```

Comment:
At this point, with the change from header separate from data to header appended to previous data, backwards compatability is not possible. Seems like a good time for cleanup.

Reply:
According to @fritzm, any changes in the Protobuf attributes require changing the numbers. This will prevent mysterious crashes in case if (intentionally or accidentally) different versions of the Protobuf schema will be deployed at a client and a server. I think what's done here was the right thing to do.

Reply:
There is a difference between having mismatched versions accidentally running at the same time causing clean error fails, and having mismatched versions accidentally running at the same time causing exploding segvs of death...

Also, the point is not that numbers have to _change_, but that they should _never be recycled_...

Reply:
I think that's a change on arguments I've heard in the past, where there was some hope of backward compatibility. There's no hope of backward compatibility in this change. Hoping for backwards compatibility seems risky to me.

The new argument seems to be to never reuse numbers, so it fails loudly an obviously when there's a protocol issue between worker/czar. That seems safer.

There is a small benefit to using small numbers as those <=15 only use one byte to encode, so the messages are smaller. This may actually matter when dealing with result rows.

Reply:
I've removed several required fields and put in checks to make sure that they are in the received messages. Most were fields added in the tickets related to this ticket, but some were older. It still seems like a good time for house cleaning.